### PR TITLE
CSI secrets for addons

### PR DESCRIPTION
**Pull request by shapirov103**
*Issue #, if available:*

*Description of changes:*
Changes to allow add-ons to leverage secret store for sensitive data. Some refactoring, naming was also applied. Extensibility guide updated. After this change is merged, I will update the extension example as well. Currently the secret use in add-ons is on a branch [here](https://github.com/shapirov103/ssp-eks-extension/blob/feature/csi-example/lib/index.ts).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/addons/secrets-store/csi-driver-provider-aws-secrets.ts [hunk 1]
```diff
@@ -1,199 +1,239 @@
...
+    DOCKER_CONFIG = 'kubernetes.io/dockercfg',
+    SSH_AUTH = 'kubernetes.io/ssh-auth',
+    SERVICE_ACCOUNT_TOKEN = 'kubernetes.io/service-account-token',
+    TLS = 'kubernetes.io/tls'
 }
 
 interface ParameterObject {
     objectName: string;
     objectType: string;
+    jmesPath?: JmesPathObject[];
```
Comment (shapirov103) [inline lib/addons/secrets-store/csi-driver-provider-aws-secrets.ts:96]: Needed for complex secrets (e.g. JSON structure)

**Review by askulkarni2 — APPROVED** [review]
Great PR!

**Review by youngjeong46 — APPROVED** [review]
LGTM.

**Pull request merged**